doc/rst/language/spec/data-parallelism.rst

```diff
@@ -752,40 +752,28 @@ side array expressions alias the left-hand side expression.
    are assigned to ``A`` may be read to compute the sum depending on the
    number of tasks used to implement the data parallel statement.
 
-.. _Indirect_Whole_Array_Indexing:
-
-Indirect Whole Array Indexing
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Arrays can be indexed indirectly using an array of indices.
-Given an array ``A`` and an array of indices ``B`` in the domain of ``A``,
-the function call
+Array indexing operations can also be promoted.
+For example, an array of indices can be used to index into another array
 
 .. code-block:: chapel
 
-   f(A[B]);
+   A[B];
 
-is equivalent to
+which results in the promoted expression
 
 .. code-block:: chapel
 
-   [b in B] f(A[b]);
+   [b in B] A[b];
 
-This is a legal expression only if the result of the promotion ``A[B]`` is not modified.
-The following statement
+However, it is an error to modify this promoted expression.
+For example, the following is an error
 
 .. code-block:: chapel
 
    A[B] += 3;
 
-is equivalent to
-
-.. code-block:: chapel
-
-   [b in B] A[b] += 3;
-
-which is not legal, as ``A`` cannot be modified without an explicit ``ref`` shadow variable.
```

Comment:
I thought this rationale was strong and would restore it.  Without it, we seem to rely on the user snapping to why this would be on their own, which it isn't obvious they will.

Reply:
I have restored this